geomesa-fs/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/ParquetFileSystemStorage.scala

```diff
@@ -120,42 +122,60 @@ class ParquetFileSystemStorage(root: Path,
   override def listPartitions(typeName: String): util.List[String] =
     metadata(typeName).getPartitions
 
-  // TODO ask the parition manager the geometry is fully covered?
+  // TODO ask the partition manager the geometry is fully covered?
   override def getPartitionReader(sft: SimpleFeatureType, q: Query, partition: String): FileSystemPartitionIterator = {
+
+    import org.locationtech.geomesa.index.conf.QueryHints._
+    import scala.collection.JavaConversions._
+
+    // parquetSft has all the fields needed for filtering and return, returnSft just has those needed for return
+    val (parquetSft, returnSft) = q.getHints.getTransformSchema match {
```

Comment:
Is this useful elsewhere in the codebase?

Reply:
most of our data stores filter on the full feature, and then apply any projections. parquet is different because it won't even return the non-projected attributes, hence the checks here. i don't know of anywhere else it would be used at this point.